File: Python/Mahasiswa.py
# library
from SivitasAkademik import SivitasAkademik

# membuat kelas mahasiswa yang merupakan turunan kelas SivitasAkademik
class Mahasiswa(SivitasAkademik):
    # atribut privat kelas Mahasiswa
    __nim = 0
    __fakultas = ""

    # konstruktor dengan parameter
    def __init__(self, nik, nama, jenkel, asal_universitas, email_edu, nim, fakultas):
        super().__init__(nik, nama, jenkel, asal_universitas, email_edu)
        #set nilainya
        self.__nim = nim
        self.__fakultas = fakultas
    
    # tetapkan nim
    def setNim(self, nim):
        self.__nim = nim
    
    # get nim
    def getNim(self):
        return self.__nim
    
    # tetapkan fakultas
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas
    
    # get fakultas
    def getFakultas(self):
        return self.__fakultas

File: CPP/Mahasiswa.cpp
// library
#include "SivitasAkademik.cpp"
#ifndef Mahasiswa_H
#define Mahasiswa_H

// membuat kelas mahasiswa yang merupakan turunan dari kelas SivitasAkademik
class Mahasiswa: public SivitasAkademik {
    // atribut yang diperlukan
    private:
        int nim;
        string fakultas;
    
    public:
        // konstruktor kosong
        Mahasiswa() {
        }

        // konstruktor dengan parameter
        Mahasiswa(int nim, string nama, string jenkel, string fakultas) {
            // set nilainya
            this->nim = nim; 
            this->nama = nama;
            this->jenkel = jenkel;
            this->fakultas = fakultas;
        }

         // tetapkan nim
        void setNim(int nim) {
            this->nim = nim;
        }

        // get nim
        int getNim() {
            return this->nim;
        }

         // tetapkan fakultas
        void setFakultas(string fakultas) {
            this->fakultas = fakultas;
        }
        
        // get fakultas
        string getFakultas() {
            return this->fakultas;
        }
        
        // deconstruct
        ~Mahasiswa() {
        }
};

#endif